pkg/webhook/mutation.go

```diff
@@ -0,0 +1,228 @@
+func getPebbleVolumeMountPatches(pod *corev1.Pod) []jsonpatch.Operation {
+	patches := []jsonpatch.Operation{}
+
+	for i, container := range pod.Spec.Containers {
+		// We don't need to mount a volume if the root FS is not read-only.
+		if !containerNeedsPebbleVolume(container) {
+			continue
+		}
+
+		// Make sure that there's no volume already targeting the $PEBBLE path.
+		// If there is, we'll let the user handle it.
+		// We might want to check if the user also defined the PEBBLE_COPY_ONCE env variable
+		// in this case, as the $PEBBLE folder won't have the layers folder needed by Pebble.
+		pebblePath := getContainerPebblePath(container)
+		if containerHasMountPath(container, pebblePath) {
+			continue
+		}
+
+		// The layers folder exists in the $PEBBLE path. This means we can't mount there, as
+		// that will cause the layers folder to no longer exists in the $PEBBLE path.
+		// Instead, we should mount in a subfolder, and set the $PEBBLE and $PEBBLE_READ_ONCE env vars.
+		mountPath := filepath.Join(pebblePath, PEBBLE_WRITABLE_SUBPATH)
+
+		// Add volume patch to the current container.
+		// The subpath is required if there are multiple rocks in the same Pod.
+		patches = append(patches, jsonpatch.NewPatch("add", fmt.Sprintf(CONTAINER_VOLUME_MOUNT_PATH, i),
+			corev1.VolumeMount{
+				Name:      "pebble-dir",
+				MountPath: mountPath,
+				SubPath:   container.Name,
+			},
+		))
+
+		// We're adding the same volume mount to all the containers in the Pod.
+		// If we have multiple rocks in the same Pod, and they have the same $PEBBLE path,
+		// they'd end up using the same socket and state files. We have to prevent that.
+		patches = append(patches, getContainerEnvPatchOps(container, pebblePath, mountPath, i)...)
+	}
+
+	// If we don't have any volume mounts, we don't need to add any volume.
+	if len(patches) == 0 {
+		return patches
+	}
+
+	// Add patch for Pebble volume.
+	patches = append(patches, jsonpatch.NewPatch("add", POD_VOLUME_PATH,
+		corev1.Volume{
+			Name:         "pebble-dir",
+			VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}},
+		},
+	))
+
+	return patches
+}
+
+func getPod(ar *v1.AdmissionReview) (*corev1.Pod, error) {
+	pod := corev1.Pod{}
+
+	if _, _, err := deserializer.Decode(ar.Request.Object.Raw, nil, &pod); err != nil {
+		return nil, fmt.Errorf("Request could not be decoded: %v", err)
+	}
+
+	return &pod, nil
+}
+
+// Check if the given container has the given mount path.
+// Add an empty dir volume for Pebble to store its state in.
+func addPebbleMountMutation(ar *v1.AdmissionReview) (*v1.AdmissionResponse, error) {
+	log.Debug("calling addPebbleMountMutation")
+
+	podResource := metav1.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+	if ar.Request.Resource != podResource {
+		err := fmt.Errorf("Expected resource to be: '%s', actual: '%s'", podResource, ar.Request.Resource)
+		return nil, err
```

Comment:
just directly return the error and please return errors in lowercase https://staticcheck.dev/docs/checks#ST1005

Reply:
Done